configs/plugins/mermaid-reporter-plugin.js

```diff
@@ -0,0 +1,111 @@
+const figures = require("figures");
+
+/* eslint-disable security/detect-object-injection */
+const mermaidNode = (pNode, pText) => {
+  const lNode = pNode
+    .replace(figures.cross, "__unknown__")
```

Comment:
The ✖ is emitted by an upstream dependency (acorn-loose - javascript parser) - and only when it can make [heads nor tails of something](https://github.com/acornjs/acorn/tree/master/acorn-loose#interface). I've gone to some lengths to make that happen as little as possible :-).

Reply:
On windows (CI) this appears to fail.

This is probably because figures uses a fallback on windows [(`×`)](https://github.com/sindresorhus/figures/blob/main/index.js#L257). As acorn-loose doesn't use `figures.cross` but [instead seems to use the literal '✖' character](https://github.com/acornjs/acorn/blob/fb4afdecf7ff588cc65ddef608d3e6e88e05edb5/acorn-loose/src/parseutil.js#L1). 

It'll probably pan out OK windows if we started using it in that fashion as well.

Reply:
I've added a commit (38034c8) to the PR to verify the assumption above - and it seems to work as expected...

Reply:
I see, so it falls back on Windows👀
And if it's not ✖️, mermaid compiles through, so the commit you added looks good!

Reply:
An additional thought: the replacement strategy we employ here will probably cover most use cases, even though it isn't water tight - partly because mermaid syntax has some interesting properties (e.g. `a-b` is valid identifier. `a--b` is not, and `a---b` is two identifiers that have a relationship; this while all three examples are valid file names). So maybe in the future we'll need to revise our strategy - e.g. by using a hash function. That would hamper the legibility of the generated .mmd source, though. 

I'll merge this as an MVP.

Reply:
> this while all three examples are valid file names

It certainly has the potential to break... Exploring edge cases is helpful, thanks.

I thought that if we could only deal with such filenames, we could preserve the readability of the mmd files, but if I could get `mermaid.parse` working in node first, we would have more options. I don't know if it will work, but I hope to try it soon.